hazelcast-client/src/test/java/com/hazelcast/client/ClientServiceTest.java

```diff
@@ -536,4 +536,20 @@ public void run() throws Exception {
             }
         }, 4);
     }
+
+    @Test
+    public void testAddingListenersOpenConnectionsToAllCluster() {
```

Comment:
can you also test this: one member cluster, then client starts and gets connected to the cluster. Then start other 3 members in three different threads and addListener on the main test thread.

Reply:
How am I going to verify different threads ? We don't support sync in case of different threads anyway.